src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,35 @@
 package core.basesyntax;
 
-public class WorkWithFile {
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
 
+public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
-
+        try (FileWriter fileWriter = new FileWriter(toFileName)) {
+            if (fromFileName == null) {
+                throw new RuntimeException("No such file");
+            }
+            List<String> stringList = Files.readAllLines(Paths.get(fromFileName));
+            int supplyCount = 0;
+            int buyCount = 0;
+            int resultCount;
+            for (String line : stringList) {
+                if (line.startsWith("supply")) {
+                    supplyCount += Integer.parseInt(line.replaceAll("\\D+",""));
```

Comment:
- look through checklist and try to split logic into several small methods
- use constants for magic numbers and regexes
- use english for commit and PR names